ertcrypto/ertcrypto.go

```diff
@@ -72,7 +73,17 @@ func SealWithProductKey(plaintext []byte) ([]byte, error) {
 
 // Unseal decrypts data encrypted with the unique or product CPU key with the help of the embedded key info
 func Unseal(ciphertext []byte) ([]byte, error) {
+	if len(ciphertext) < 4 {
+		return nil, fmt.Errorf("ciphertext is too short")
+	} 
+
 	keyInfoLength := binary.LittleEndian.Uint32(ciphertext[:4])
+	
+	// We might deal with invalid user data as input, so let's convert an potential upcoming out-of-bounds panic to an error the underlying caller can choose how to deal with this situation.
+	if 4+int(keyInfoLength) > len(ciphertext) {
```

Comment:
```suggestion
	if keyInfoLength <= 0 || 4+int(keyInfoLength) > len(ciphertext) {
```
and add `Unseal([]byte{0, 0, 0, 0})` to the test.
Also add and fix `Unseal([]byte{4, 0, 0, 0, 'i', 'n', 'f', 'o'})`.

Reply:
<= 0 instead of == 0? It's an uint32

Reply:
both is fine with me

Reply:
Went with ==0. If we actually end up with a negative number in an uint, we properly should panic as the environment became insane anyway :)